### PR TITLE
To-do list Testing

### PR DESCRIPTION
**Pull request by AkintoyeOlamide**
This is a test for methods in the JS files, to determine if some things are working well. Positive and Negative tests were processed.

### CODE REVIEW

**Comment by rukundoeric** [comment]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - Follows gitwork-flow. 💯 
  - All tests are passing. 💯

- ### To Improve
  - [ ] [MANDATORY] In your README, add instructions that need to be followed to set up and run tests locally.


_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

File: testing/index.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  priority: {
+    value: ' ',
+  },
+};
+
+const clearForm = () => {
+  todoForm.title.value = '';
+  todoForm.description.value = '';
+  todoForm.date.value = '';
+};
```
Comment (SajjadAhmad14) [inline testing/index.test.js:20]: You should not write functions again here. You can import these functions from

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [ ] You have mastered Git flow.
- [ ] README is professional and includes all necessary info to run the project locally.

## To Improve 🙏
- [ ] You should write tests for all the methods. Please add negative tests as well.
- [ ] You are writing todoList and todoForm functions again in the todo.test.js file. You should import them from the relevant file.

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, thank you for working on changes suggested previously :ok_hand:

## To Improve 🙏
- [ ] You are still defining #todoList and #createTodo methods again in todo.test.js. Kindly import them from the relevant module.
- [ ] Please add tests for all the methods. Moreover, add negative tests.
- [ ] You can use beforeEach which is the built-in Jest method if you want to define a variable or object which is used repeatedly. Please see this [link](https://jestjs.io/docs/setup-teardown).
- [ ] Please remove commented code.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

File: testing/todo.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+const todoList = (title, description, dueDate, priority) => ({
+  title,
+  description,
+  dueDate,
+  priority,
+});
```
Comment (rindrajosia) [inline testing/todo.test.js]: - [ ] [REQUIRED] Please import this function instead of defining it in this file

File: testing/todo.test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    value: 'low',
+  },
+};
+const createTodo = () => {
+  const title = todoForm.title.value;
+  const description = todoForm.description.value;
+  const dueDate = todoForm.date.value;
+  const priority = todoForm.priority.value;
+  return todoList(title, description, dueDate, priority);
+};
```
Comment (rindrajosia) [inline testing/todo.test.js]: - [ ] [REQUIRED] Please import this function instead of defining it in this file

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- No commented code :+1: 

To improve :factory_worker: [REQUIRED]:

- [ ]  You are still defining #todoList and #createTodo methods again in todo.test.js. Kindly import them from the relevant module.
- [ ] Please add tests for all the methods. Moreover, **add negative tests.**

Nice to have [OPTIONAL] :

- [ ]  

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi @AkintoyeOlamide , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
- [ ] [Optional] You have no PR message. Please add a proper PR message. 
- [ ] Please make the changes requested by previous TSEs as I can see you haven't made any change. 

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Best, 
Arslan Bisharat
slack: arslanbisharat

**Review by arslanbisharat — APPROVED** [review]
#Approved 👏 

Hi @AkintoyeOlamide, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**